File: src/main/java/me/tajam/jext/command/ExecutorDiscGive.java
package me.tajam.jext.command;

import me.tajam.jext.config.ConfigDiscManager;
import me.tajam.jext.DiscContainer;
import me.tajam.jext.Log;

import java.util.List;

import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

class ExecutorDiscGive extends ExecutorAdapter {

  ExecutorDiscGive() {
    super("discgive");
    addParameter(new ParameterPlayer(true));
    addParameter(new ParameterDisc(true));
  }

  @Override
  boolean executePlayer(Player sender, String[] args) {
    return mergedExecute(sender, args);
  }

  @Override
  boolean executeCommand(CommandSender sender, String[] args) {
    return mergedExecute(sender, args);
  }

  private boolean mergedExecute(CommandSender sender, String[] args) {
    final List<Player> players = new PlayerSelector(sender, args[0]).getPlayers();
    if (players == null) return true;

    final DiscContainer disc = ConfigDiscManager.getInstance().getDisc(args[1]);
    if (disc == null) {
      new Log().eror().t("Disc with the namespace ").o(args[1]).t(" doesn't exists.").send(sender);
      return true;
    }

    for (Player player : players) {
      player.getInventory().addItem(disc.getDiscItem());
      new Log().info().t("Received ").p().t(" disc.").send(player, disc);
    }

    final Integer playerCount = players.size();
    if (playerCount >= 2) {
      new Log().warn().t("Given ").o().t(" disc to ").o().t(" players!").send(sender, disc, playerCount);
    } else if (playerCount == 1) {
      new Log().okay().t("Given ").o().t(" disc to ").o(players.get(0).getDisplayName()).t(".").send(sender, disc);
    } else {
      new Log().eror().t("Given disc to no player, something might when wrong!").send(sender);
    }
    return true;
  }
}

File: src/main/java/me/spartacus04/jext/command/ExecutorDiscGive.kt
package me.spartacus04.jext.command

import me.spartacus04.jext.command.adapter.ExecutorAdapter
import me.spartacus04.jext.command.adapter.ParameterDisc
import me.spartacus04.jext.command.adapter.ParameterPlayer
import me.spartacus04.jext.config.sendJEXTMessage
import me.spartacus04.jext.disc.DiscContainer
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player

/**
 * ExecutorDiscGive is a class used to register the "discgive" command to the plugin.
 *
 * @constructor The constructor is empty because the class does not have any properties.
 */
internal class ExecutorDiscGive : ExecutorAdapter("discgive") {
    init {
        addParameter(ParameterPlayer(true))
        addParameter(ParameterDisc(true))
    }

    /**
     * The function `execute` gives the selected players a disc.
     *
     * @param sender The sender who executed the command.
     * @param args The arguments that were passed to the command.
     */
    override fun execute(sender: CommandSender, args: Array<String>) {
        val players = ParameterPlayer.getPlayers(args[0], sender)

        val disc = ParameterDisc.getDisc(args[1])

        if (disc == null) {
            sender.sendJEXTMessage("disc-namespace-not-found", hashMapOf(
                "namespace" to args[1]
            ))

            return
        }

        for (player in players) {
            player.inventory.addItem(DiscContainer(disc).discItem)

            sender.sendJEXTMessage("disc-received", hashMapOf(
                "disc" to disc.TITLE
            ))
        }

        val playerCount = players.size

        if (playerCount >= 2) {
            sender.sendJEXTMessage("disc-given-multiple", hashMapOf(
                "disc" to disc.TITLE,
                "playercount" to playerCount.toString()
            ))
        } else if (playerCount == 1) {
            sender.sendJEXTMessage("disc-given", hashMapOf(
                "disc" to disc.TITLE,
                "player" to players[0].name
            ))
        } else {
            sender.sendJEXTMessage("no-disc-given")
        }
    }
}